frontend/src/lib/hooks.ts

```diff
@@ -243,19 +267,41 @@ export const useWrite = <
   return useMutation({
     mutationKey: [type],
     mutationFn: (params: P) => komodo_client().write<T, R>(type, params),
-    onError: (e: { result: { error?: string; trace?: string[] } }, v, c) => {
-      console.log("Write error:", e);
+    onError: (e: { result: { error?: string; trace?: string[] }; isValidationError?: boolean; status?: number; response?: { status?: number } }, v, c) => {
       const msg = e.result.error ?? "Unknown error. See console.";
+      
+      // Log appropriately based on error type
+      if (e.isValidationError) {
+        // Log validation errors as warnings (expected user errors)
+        console.warn("Resource validation error:", e);
+      } else {
+        // Log unexpected system errors
+        console.error("Write error:", e);
+      }
+      
       const detail = e.result?.trace
         ?.map((msg) => msg[0].toUpperCase() + msg.slice(1))
         .join(" | ");
-      let msg_log = msg ? msg[0].toUpperCase() + msg.slice(1) + " | " : "";
-      if (detail) {
-        msg_log += detail + " | ";
+      
+      // Use appropriate messages for different error types
+      let toastTitle: string;
+      let toastDescription: string;
+      
+      if (e.isValidationError) {
+        toastTitle = "Invalid Input";
+        toastDescription = msg || "Please check your data and try again.";
+      } else {
+        let msg_log = msg ? msg[0].toUpperCase() + msg.slice(1) + " | " : "";
+        if (detail) {
+          msg_log += detail + " | ";
+        }
+        toastTitle = `Write request ${type} failed`;
+        toastDescription = `${msg_log}See console for details`;
```

Comment:
I don't think these changes are necessary, it should be kept close to the api error handling and better error messages should be handled by good top level  `.context()` message on the backend.